docroot/modules/custom/mass_caching/mass_caching.module

```diff
@@ -72,3 +73,17 @@ function mass_caching_path_update(array $path) {
     \Drupal::service('manual_purger')->purgePath($path['alias']);
   }
 }
+
+/**
+ * Implements hook_entity_update().
+ *
+ * Purge media's path when the entity is updated.
+ */
+function mass_caching_entity_update(EntityInterface $entity) {
```

Comment:
I know this sounds ridiculous, but can we also add `hook_entity_insert` (or `hook_media_insert`) as well?  I'm thinking of the case where an editor gets frustrated for some reason and renames an existing media entity, then creates a new one with the old title.